examples/example_weather_agent.py

```diff
@@ -0,0 +1,268 @@
+        
+    if function_result and function_result.get('status') == 'success':
+        current_state['successes'] += 1
+    elif function_result:
+        current_state['failures'] += 1
+        
+    current_state['requests'] += 1
+    return current_state
+
+def create_weather_agent() -> Agent:
+    """Create and configure a weather agent.
+    
+    This function sets up an agent with a weather worker that can
+    fetch weather information and provide recommendations.
+    
+    Returns:
+        Agent: Configured weather agent
+        
+    Raises:
+        ValueError: If VIRTUALS_API_KEY is not set
+        ValidationError: If worker configuration is invalid
+        APIError: If agent creation fails
+        
+    Example:
+        >>> agent = create_weather_agent()
+        >>> agent.compile()
+    """
+    # Get API key
+    api_key = os.getenv('VIRTUALS_API_KEY')
+    if not api_key:
+        raise ValueError("VIRTUALS_API_KEY not set")
+    
+    logger.debug("Starting weather reporter creation")
+    logger.debug(f"Creating agent with API key: {api_key[:8]}...")
+    
+    # Create weather function
+    weather_fn = Function(
+        fn_name="get_weather",
+        fn_description="Get weather information for a city",
+        executable=get_weather,
+        args=[
+            Argument(
+                name="city",
+                type="string",
+                description="City to get weather for"
+            )
+        ]
+    )
+    
+    # Create worker config
+    worker_config = WorkerConfig(
+        id="weather_worker",
+        worker_description="Provides weather information and recommendations",
+        get_state_fn=get_state,
+        action_space=[weather_fn],
+        instruction="Fetch weather data and provide clothing recommendations",
+        api_key=api_key
+    )
+    
+    # Create agent
+    agent = Agent(
+        api_key=api_key,
+        name="Weather Reporter",
+        agent_description="Reports weather and provides recommendations",
+        agent_goal="Help users prepare for weather conditions",
+        get_agent_state_fn=get_state,
+        workers=[worker_config]
+    )
+    
+    # Compile agent
+    agent.compile()
+    logger.info("Created weather reporter agent")
+    return agent
+
+def test_weather_agent(agent: Agent):
+    """Test the weather agent with different cities.
+    
+    This function runs a series of tests to verify that the agent
+    can correctly fetch and process weather information.
+    
+    Args:
+        agent (Agent): Weather agent to test
+        
+    Raises:
+        AssertionError: If any test fails
+        
+    Example:
+        >>> agent = create_weather_agent()
+        >>> test_weather_agent(agent)
+        ✨ All weather reporter tests passed!
+    """
+    logger.debug("Starting weather reporter tests")
+    
+    # Test cities
+    test_cases = ["New York", "Miami", "Boston"]
+    
+    for city in test_cases:
```

Comment:
Consider using `@pytest.mark.parameterize` to specify test cases.

Reply:
Actually, please disregard my previous comments about using pytest. Maybe it's better to _not_ assume any given test framework.